### PR TITLE
fixes button alignment and missing icons issues with ads welcome page

### LINKED ISSUE #10427: Welcome Page New/Open File Button Text Misaligned
This is what I see from my Mac:

image

Notice that 'Open File' is much higher up on the button than the 'New' text.

### LINKED ISSUE #10619: A11y_AzureDataStudioStayHealthy_Welcome_AzureDataStudio-Insiders_AI4D: All controls are set as links.
[“Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags:#A11y_AzureDataStudioStayHealthy;#A11ySev2;#A11yTCS;#A11yMAS;#Benchmark;#DesktopApp;#MAS4.1.2;#MAS1.3.1;#SQL Azure Data Studio;#FTP;#AI4D;#MAS4.1.2;#Win32;

**Environment Details:**
Application Name: Azure Data Studio
Version: 1.19.0-insider
Commit: e8dc0d1
Date: 2020-05-27T17:24:57.733Z
VS Code: 1.46.0
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

**Additional Details:**
MAS References:MAS4.1.2, MAS1.3.1, MAS4.2.1
Tool: Accessibility insights for desktop

**Repro Steps:**
1 .Launch Azure Data Studio Insiders application.
2 . If Welcome screen is not shown on application launch, then navigate to Help --> Welcome.
3 . Launch accessibility insights for desktop and hover over the controls on the screen.

**Actual:**
Issue1:
On hovering over all the controls all of them are set as links, but on clicking each of them new dialog or tab have opened no redirection to a website.

Issue2:
On hovering over the new button the control is set as group and no name.

**Same issue in the following screen:**
1 .Resources-show more control 

**Expected:**
Issue1:
The control type for all controls should be set as a button and same should be announced by screen readers.
Since on activating each control, there is no redirection to a new url in the web browser.

Issue2:
The new control should be set as button with name as "New"

**User Impact:**
Screen reader users will get wrong control information of the control.

**Attachment link for Reference:**
A11y_AzureDataStudioStayHealthy_Welcome_AzureDataStudio-Insiders_AI4D
A11y_AzureDataStudioStayHealthy_Welcome_AzureDataStudio-Insiders_AI4D_Issue2

### LINKED ISSUE #10642: A11y_AzureDataStudioStayHealthy_Welcome_ExtendyourdataStudio_Keyboard:The install button tool tip is not hoverable.
[“Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags:#A11y_AzureDataStudioStayHealthy;#A11ySev2;#A11yTCS;#A11yWCAG2.1;#NonBenchmark;#DesktopApp;#MAS1.4.13;#SQL Azure Data Studio;#FTP;#Keyboard;#STP;#Win32;

**Environment Details:**
Application Name: Azure Data Studio
Version: 1.19.0-insider
Commit: e8dc0d1
Date: 2020-05-27T17:24:57.733Z
VS Code: 1.46.0
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

**Additional Details:**
MAS References:MAS1.4.13

**Repro Steps:**
1 . Launch Azure Data Studio Insiders application.
2 . If Welcome screen is not shown on application launch, then navigate to Help --> Welcome.
3 . Navigate Extend your data Studio on Welcome screen.
4 . Navigate to install button 

**Actual:**
On hovering over the button install the tooltip appears but on moving to the tooltip it disappears.

**Expected:**
When hovered over the tooltip the tooltip should be hoverable and not dismiss.

**User Impact**:
Keyboard user on trying to move to the tooltip won't be able to view it since it disappears.

**Attachment link for Reference:**
[10642A11y_AzureDataStudioStayHealthy_Welcome_ExtendyourdataStudio_Keyboard.zip](https://github.com/microsoft/azuredatastudio/files/4705161/10642A11y_AzureDataStudioStayHealthy_Welcome_ExtendyourdataStudio_Keyboard.zip)

### LINKED ISSUE #10696: Buttons text not aligned in Welcome Page
The text in the buttons Welcome Page are not vertically aligned properly.

Screen Shot 2020-06-03 at 10 47 43 AM

### LINKED ISSUE #10731: icon is missing for powershell and postgres on welcome page
Screen Shot 2020-06-04 at 10 51 17 AM

### LINKED ISSUE #11205: Extensions in welcome page no icon
Issue Type: <b>Bug</b>

The icons for some of the extensions in the welcome page are blank.

image

Azure Data Studio version: azuredatastudio-insiders 1.20.0-insider (2d989e1180f342d684a4e26f76b941843b7fc261, 2020-07-03T05:33:38.245Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor             (24 x 3793)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.95GB (7.72GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
azuredatastudio-sqlite|ant|1.0.7
material-icon-theme|PKi|3.9.2
vscode-icons|vsc|9.6.0

</details>

### PR DESCRIPTION
**Pull request by v-bbrady**
This fixes issues #10696, #11205, #11205, #10619 and  #10642 and #10427. It fixes the button alignment issues on the welcome page by changing them over from a DOM created element to using the button component that exists with in ADS. It also fixes the issue of the missing icons on the extension tiles.

image

image

This PR fixes #10696 
This PR fixes #10642
This PR fixes #10427
This PR fixes #10731 
This PR fixes #11205
This PR fixes #10619

### CODE REVIEW

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.css [hunk 1]
```diff
@@ -144,28 +144,7 @@
...
-	display: flex;
-	align-content: center;
-	justify-content: center;
-	text-align: center;
-	text-decoration: none;
-	margin: 10px 0
-}
-
-.ads_homepage.SM .btn {
+/* .ads_homepage.SM .btn {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.css:147]: Just delete if you don't need this anymore

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.css [hunk 2]
```diff
@@ -426,15 +408,14 @@
 	mask-size: 12px 12px
 }
 
+/*
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.css:411]: Just delete if you don't need anymore

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.css [hunk 3]
```diff
@@ -504,7 +492,7 @@
 	display: inline-block;
 	width: 11px;
 	height: 6px;
-	content: "";
+	/* content: ""; */
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.css:495]: delete

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 4]
```diff
@@ -614,38 +674,48 @@ class WelcomePage extends Disposable {
...
-				a.title = extension.title || (extension.isKeymap ? localize('welcomePage.installKeymap', "Install {0} keymap", extension.name) : localize('welcomePage.installExtensionPack', "Install additional support for {0}", extension.name));
-				a.classList.add('installExtension');
-				a.setAttribute('data-extension', extension.id);
-				a.href = 'javascript:void(0)';
-				a.addEventListener('click', e => {
+		if (btnContainer) {
+			extensionPacks.forEach((extension) => {
+
+				const installText = localize('welcomePage.install', "Install");
+				let dropdownBtn = new Button(btnContainer);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:683]: Make sure to register buttons to the disposable store (this._register)

**Comment by alanrenmsft** [comment]
@v-bbrady #10731 is due to temporary network issue that i had, i have closed it 5 minutes after its opened 😃 i don't think you need to include the icons in the source code.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.css (hunk 1 above)
Comment (alanrenmsft) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.css:147]: please remove if not necessary.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 5]
```diff
@@ -316,11 +317,70 @@ class WelcomePage extends Disposable {
...
+
+		const fileBtnContainer = document.querySelector('#open_file_btn_container') as HTMLElement;
+		const openFileText = localize('welcomePage.openFile', "Open file");
+		let openFileMac = new Button(fileBtnContainer);
+		let openFileWindows = new Button(fileBtnContainer);
+		openFileWindows.label = openFileText;
+		openFileMac.label = openFileText;
+		dropdownBtn.label = newText;
+
+		const getNewFileBtnWindows = document.querySelector('#open_file_btn_container .monaco-button:first-of-type') as HTMLAnchorElement;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:373]: Is there a reason you wouldn't just make a single button and set the appropriate values based on the platform?

**Comment by Charles-Gagnon** [comment]
@alanrenmsft That being said - I would generally say we shouldn't rely on icons from the web. There's a number of reasons that could fail and we should avoid having that kind of reliance on a connection to the internet to display things as basic as static icons.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts (hunk 5 above)
Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:373]: Hey @Charles-Gagnon I was just basing this on how VS code has theirs. I'm not sure how to set the values based on platform. Could you please advise? Thank you.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.css (hunk 1 above)
Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.css]: removed

Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.css]: @alanrenmsft removed

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.css (hunk 2 above)
Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.css]: @Charles-Gagnon deleted

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts (hunk 5 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:373]: https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/common/platform.ts#L107

Just switch on whether it's mac or not and then fill in the appropriate values based on that.

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts (hunk 4 above)
Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:681]: done

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts (hunk 5 above)
Comment (v-bbrady) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts]: okay I think it's okay now

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts (hunk 4 above)
Comment (alanrenmsft) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:681]: @Charles-Gagnon is adding button to the disposable store enough? should we also add the event handler, right?

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:681]: Well, for buttons we should be using the onDidClick event https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/browser/ui/button/button.ts#L47 which is registered with the button.

@v-bbrady You should be using that instead of setting the href and other stuff directly. 

And for this whole dropdown thing - Hale added a DropdownMenuActionViewItem recently. You should look at reusing that instead of reimplementing (yet another) dropdown menu.

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:681]: You also missed registering this button on line 681

File: src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 6]
```diff
@@ -914,23 +982,23 @@ registerThemingParticipant((theme, collector) => {
 	}
 	const buttonHoverBackgroundColor = theme.getColor(buttonHoverBackground);
 	if (buttonHoverBackgroundColor) {
-		collector.addRule(`.monaco-workbench .part.editor > .content .welcomePageContainer .btn_primary:hover { background: ${buttonHoverBackgroundColor};}`);
+		collector.addRule(`.monaco-workbench .part.editor > .content .welcomePageContainer .btn_primary:hover { background: ${buttonHoverBackgroundColor} !important;}`);
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts:985]: usually we will try to avoid using !important, why are we adding that here in this PR? that doesn't seem to be related to the issues we are fixing.

**Comment by Charles-Gagnon** [comment]
@v-bbrady Can you get this synced up to the latest from main so we can review these changes again with your latest fixes?

**Comment by v-bbrady** [comment]
> 
> 
> @v-bbrady Can you get this synced up to the latest from main so we can review these changes again with your latest fixes?

@Charles-Gagnon This has been updated to be aligned with main.

**Comment by v-bbrady** [comment]
> 
> 
> So this PR is just fixing the button alignment now right? The missing icons were already merged in an earlier PR from what I see.
